components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/common/OneTimeTriggerInboundRunner.java

```diff
@@ -0,0 +1,101 @@
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.core.multitenancy.utils.TenantAxisUtils;
+import org.wso2.carbon.inbound.endpoint.persistence.service.InboundEndpointPersistenceServiceDSComponent;
+import org.wso2.carbon.inbound.endpoint.protocol.mqtt.MqttTask;
+import org.wso2.carbon.mediation.clustering.ClusteringAgentUtil;
+import org.wso2.carbon.utils.CarbonUtils;
+import org.wso2.carbon.utils.ConfigurationContextService;
+
+public class OneTimeTriggerInboundRunner implements Runnable {
+
+    private OneTimeTriggerInboundTask task;
+    private volatile boolean execute = true;
+    private volatile boolean init = false;
+    private String tenantDomain;
+    private static final long CLUSTER_CONFIGURATION_CHECK_INTERVAL = 1000;
+
+    private static final Log log = LogFactory.getLog(InboundRunner.class);
+
+    public OneTimeTriggerInboundRunner(OneTimeTriggerInboundTask task, String tenantDomain) {
+        this.task = task;
+        this.tenantDomain = tenantDomain;
+    }
+
+    @Override
+    public void run() {
+        if (log.isDebugEnabled()) {
+            log.debug("Starting the Inbound Endpoint.");
+        }
+        // Wait for the clustering configuration to be loaded.
+        while (!init) {
+            if (log.isDebugEnabled()) {
+                log.debug("Waiting for the configuration context to be loaded to run Inbound Endpoint.");
+            }
+            Boolean isSinglNode = ClusteringAgentUtil.isSingleNode();
+            if (isSinglNode != null) {
+                if (!isSinglNode && !CarbonUtils.isWorkerNode()) {
+                    // Given node is the manager in the cluster, and not
+                    // required to run the service
+                    execute = false;
+                    log.info("Inbound Endpoint will not run in manager node. Same will run on worker(s).");
+                }
+                init = true;
+            }
+            try {
+                Thread.sleep(CLUSTER_CONFIGURATION_CHECK_INTERVAL);
+            } catch (InterruptedException e) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Unable to sleep the Inbound thread for interval of : " +
+                            CLUSTER_CONFIGURATION_CHECK_INTERVAL + "ms.");
+                }
+            }
+        }
+
+        if (log.isDebugEnabled()) {
+            log.debug("Configuration context loaded. Ready to Run the Inbound Endpoint.");
+        }
+        if (log.isDebugEnabled()) {
+            log.debug("Executing One-Time trigger connection consumer task in Inbound Endpoint.");
+        }
+
+        try {
+            if (execute) {
+                task.taskExecute();
+            }
+        } catch (Exception e) {
+            log.error("Error executing the Inbound Endpoint once time execution", e);
+        }
+        //Keep the tenant loaded
+        if (tenantDomain != null) {
```

Comment:
We may need to keep the tenant loaded. But this code will only be executed once?